src/odemis/acq/align/autofocus.py

```diff
@@ -377,10 +537,18 @@ def AutoFocus(detector, emt, focus, dfbkg=None, good_focus=None):
     f.task_canceller = _CancelAutoFocus
 
     # Run in separate thread
-    autofocus_thread = threading.Thread(target=executeTask,
-                                        name="Autofocus",
-                                        args=(f, _DoAutoFocus, f, detector, emt,
-                                              focus, dfbkg, good_focus))
+    if method == "exhaustive":
```

Comment:
As all the functions take the same parameters, you can simplify a bit the code by just setting a "autofocus_fn" variable with the right function, and creating the thread after the 'if'.